### ISSUE TITLE
conda list not working inside conda environment

### ISSUE THREAD

**Issue by sergiuiacob1** [conda/conda#9246]
## Current Behavior
`conda list` generates the following error:

```# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1062, in __call__
        return func(*args, **kwargs)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/cli/main_list.py", line 142, in execute
        show_channel_urls=context.show_channel_urls)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/cli/main_list.py", line 80, in print_packages
        show_channel_urls=show_channel_urls)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/cli/main_list.py", line 45, in list_packages
        installed = sorted(PrefixData(prefix, pip_interop_enabled=True).iter_records(),
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/core/prefix_data.py", line 133, in iter_records
        return itervalues(self._prefix_records)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/core/prefix_data.py", line 162, in _prefix_records
        return self.__prefix_records or self.load() or self.__prefix_records
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/core/prefix_data.py", line 72, in load
        self._load_site_packages()
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/core/prefix_data.py", line 277, in _load_site_packages
        python_record = read_python_record(self.prefix_path, af, python_pkg_record.version)
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/gateways/disk/read.py", line 257, in read_python_record
        paths_tups = pydist.get_paths()
      File "/Users/sergiuiacob/miniconda3/lib/python3.7/site-packages/conda/common/pkg_formats/python.py", line 269, in get_paths
        records = sorted(concatv(records, ((pf, None, None) for pf in missing_pyc_files)))
    TypeError: '<' not supported between instances of 'NoneType' and 'str'

`$ /Users/sergiuiacob/miniconda3/bin/conda list`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=automl
                CONDA_EXE=/Users/sergiuiacob/miniconda3/bin/conda
             CONDA_PREFIX=/Users/sergiuiacob/miniconda3/envs/automl
    CONDA_PROMPT_MODIFIER=(automl)
         CONDA_PYTHON_EXE=/Users/sergiuiacob/miniconda3/bin/python
               CONDA_ROOT=/Users/sergiuiacob/miniconda3
              CONDA_SHLVL=1
                     PATH=/Users/sergiuiacob/miniconda3/bin:/Users/sergiuiacob/miniconda3/envs/a
                          utoml/bin:/Users/sergiuiacob/miniconda3/condabin:/usr/local/bin:/usr/b
                          in:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/sergiuiacob/development/fl
                          utter/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : automl
    active env location : /Users/sergiuiacob/miniconda3/envs/automl
            shell level : 1
       user config file : /Users/sergiuiacob/.condarc
 populated config files : /Users/sergiuiacob/.condarc
          conda version : 4.7.11
    conda-build version : not installed
         python version : 3.7.4.final.0
       virtual packages : 
       base environment : /Users/sergiuiacob/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/sergiuiacob/miniconda3/pkgs
                          /Users/sergiuiacob/.conda/pkgs
       envs directories : /Users/sergiuiacob/miniconda3/envs
                          /Users/sergiuiacob/.conda/envs
               platform : osx-64
             user-agent : conda/4.7.11 requests/2.22.0 CPython/3.7.4 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.

If submitted, this report will be used by core maintainers to improve
future releases of conda.
Would you like conda to send this report to the core maintainers?

[y/N]:  
```

### Steps to Reproduce
Activate an environment and run `conda list`

## Expected Behavior
The command works without any environments activated, which is strange
<details><summary><code>`conda deactivate; conda list`</code></summary><p>

```
# packages in environment at /Users/sergiuiacob/miniconda3:
#
# Name                    Version                   Build  Channel
appnope                   0.1.0                    py37_0  
asn1crypto                0.24.0                   py37_0  
attrs                     19.1.0                   py37_1  
backcall                  0.1.0                    py37_0  
bleach                    3.1.0                    py37_0  
bzip2                     1.0.8                h1de35cc_0  
ca-certificates           2019.5.15                     1  
certifi                   2019.6.16                py37_1  
cffi                      1.12.3           py37hb5b8e2f_0  
chardet                   3.0.4                 py37_1003  
conda                     4.7.11                   py37_0  
conda-package-handling    1.3.11                   py37_0  
cryptography              2.7              py37ha12b0ac_0  
decorator                 4.4.0                    py37_1  
defusedxml                0.6.0                      py_0  
entrypoints               0.3                      py37_0  
idna                      2.8                      py37_0  
ipykernel                 5.1.2            py37h39e3cac_0  
ipython                   7.8.0            py37h39e3cac_0  
ipython_genutils          0.2.0                    py37_0  
jedi                      0.15.1                   py37_0  
jinja2                    2.10.1                   py37_0  
jsonschema                3.0.2                    py37_0  
jupyter_client            5.3.1                      py_0  
jupyter_core              4.5.0                      py_0  
libarchive                3.3.3                h786848e_5  
libcxx                    4.0.1                hcfea43d_1  
libcxxabi                 4.0.1                hcfea43d_1  
libedit                   3.1.20181209         hb402a30_0  
libffi                    3.2.1                h475c297_4  
libiconv                  1.15                 hdd342a3_7  
libsodium                 1.0.16               h3efe00b_0  
libxml2                   2.9.9                hf6e021a_1  
lz4-c                     1.8.1.2              h1de35cc_0  
lzo                       2.10                 h362108e_2  
markupsafe                1.1.1            py37h1de35cc_0  
mistune                   0.8.4            py37h1de35cc_0  
nb_conda                  2.2.1                    py37_0  
nb_conda_kernels          2.2.2                    py37_0  
nbconvert                 5.5.0                      py_0  
nbformat                  4.4.0                    py37_0  
ncurses                   6.1                  h0a44026_1  
notebook                  6.0.1                    py37_0  
openssl                   1.1.1d               h1de35cc_1  
pandoc                    2.2.3.2                       0  
pandocfilters             1.4.2                    py37_1  
parso                     0.5.1                      py_0  
pexpect                   4.7.0                    py37_0  
pickleshare               0.7.5                    py37_0  
pip                       19.2.2                   py37_0  
prometheus_client         0.7.1                      py_0  
prompt_toolkit            2.0.9                    py37_0  
ptyprocess                0.6.0                    py37_0  
pycosat                   0.6.3            py37h1de35cc_0  
pycparser                 2.19                     py37_0  
pygments                  2.4.2                      py_0  
pyopenssl                 19.0.0                   py37_0  
pyrsistent                0.14.11          py37h1de35cc_0  
pysocks                   1.7.0                    py37_0  
python                    3.7.4                h359304d_1  
python-dateutil           2.8.0                    py37_0  
python-libarchive-c       2.8                     py37_13  
pyzmq                     18.1.0           py37h0a44026_0  
readline                  7.0                  h1de35cc_5  
requests                  2.22.0                   py37_0  
ruamel_yaml               0.15.46          py37h1de35cc_0  
send2trash                1.5.0                    py37_0  
setuptools                41.0.1                   py37_0  
six                       1.12.0                   py37_0  
sqlite                    3.29.0               ha441bb4_0  
terminado                 0.8.2                    py37_0  
testpath                  0.4.2                    py37_0  
tk                        8.6.8                ha441bb4_0  
tornado                   6.0.3            py37h1de35cc_0  
tqdm                      4.32.1                     py_0  
traitlets                 4.3.2                    py37_0  
urllib3                   1.24.2                   py37_0  
wcwidth                   0.1.7                    py37_0  
webencodings              0.5.1                    py37_1  
wheel                     0.33.4                   py37_0  
xz                        5.2.4                h1de35cc_4  
yaml                      0.1.7                hc338f04_2  
zeromq                    4.3.1                h0a44026_3  
zlib                      1.2.11               h1de35cc_3  
zstd                      1.3.7                h5bba6e5_0 
```
</p></details>

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
active environment : automl
    active env location : /Users/sergiuiacob/miniconda3/envs/automl
            shell level : 1
       user config file : /Users/sergiuiacob/.condarc
 populated config files : /Users/sergiuiacob/.condarc
          conda version : 4.7.11
    conda-build version : not installed
         python version : 3.7.4.final.0
       virtual packages : 
       base environment : /Users/sergiuiacob/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/sergiuiacob/miniconda3/pkgs
                          /Users/sergiuiacob/.conda/pkgs
       envs directories : /Users/sergiuiacob/miniconda3/envs
                          /Users/sergiuiacob/.conda/envs
               platform : osx-64
             user-agent : conda/4.7.11 requests/2.22.0 CPython/3.7.4 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /Users/sergiuiacob/.condarc <==
auto_activate_base: False
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
This generates the same error as the one above
```
</p></details>

**Comment by forrestwaters** [comment]
Fixed by https://github.com/conda/conda/pull/9132. Can you try updating to the latest version of conda available on the conda-canary channel (version 4.7.12)?
https://www.anaconda.com/what-conda-canary/

**Comment by sergiuiacob1** [comment]
> Fixed by #9132. Can you try updating to the latest version of conda available on the conda-canary channel (version 4.7.12)?
> https://www.anaconda.com/what-conda-canary/

Confirming that it works with 4.7.12
Issue closed :)
